src/backend/renderer/element/mod.rs

```diff
@@ -89,6 +96,50 @@ pub enum UnderlyingStorage<'a, R: Renderer> {
     External(&'a R::TextureId),
 }
 
+/// Defines the possible states of an element
+/// after rendering
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RenderElementState {
+    /// The element was rendered
+    Rendered,
+    /// The element was directly scanned out
+    ScannedOut,
+    /// The element was skipped
+    Skipped,
+}
+
+/// Report for an output render operation
+#[derive(Debug, Clone)]
+pub struct OutputRenderReport {
+    /// Holds the damage for presenting the bound buffer
+    pub damage: Option<Vec<Rectangle<i32, Physical>>>,
+    /// Holds the render states of the elements
+    pub element_render_states: HashMap<Id, RenderElementState>,
+}
+
+impl OutputRenderReport {
+    /// Returns whether the report contains damage
+    pub fn has_damage(&self) -> bool {
+        self.damage.is_some()
+    }
+
+    /// Return the [`RenderElementState`] for the specified [`Id`]
+    ///
+    /// Return `None` if the element was not part of the output render operation
+    pub fn element_render_state(&self, id: impl Into<Id>) -> Option<RenderElementState> {
+        self.element_render_states.get(&id.into()).copied()
+    }
+
+    /// Returns whether the element with the specified id was presented
+    ///
+    /// Returns `false` if the element with the id was not found or skipped
+    pub fn element_was_presented(&self, id: impl Into<Id>) -> bool {
```

Comment:
This name is confusing IMO. Presented should mean exactly that the state is ScannedOut. I think just `element_was_skipped` should accomplish the same thing afaict.

Reply:
Same as above, the state represents the rendering and not the scan out, again probably poorly named by me. The check should explicitly return true if the element was not skipped irregardless if the element has been sampled or assigned to a plane.